server/src/main/java/org/opensearch/cluster/service/MasterThrottlingStats.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.service;
+
+import org.opensearch.common.metrics.CounterMetric;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Contains stats of Master Task Throttling.
+ * It stores the total cumulative count of throttled tasks per task type.
+ */
+public class MasterThrottlingStats {
```

Comment:
lets create a `MasterTaskThrottlerListener`like 

```
public interface MasterTaskThrottlerListener {
 void onThrottle(String type, final int permits);
}
```
and let `MasterThrottlingStats ` implement `MasterTaskThrottlerListener`

Reply:
In future we can create the Listener when we want to perform additional things on task throttle, we can do it

I think `MasterThrottlingStats` should not implement `MasterTaskThrottlerListener` as it has other exposed methods for returning the throttled metric count as well. We need to use it with own class reference only.

Reply:
@dhwanilpatel : ```MasterThrottlingStats``` can have more method besides implementing the MasterTaskThrottlerListener interface like you mentioned for returning metric counts.

Reply:
Sure, created Listener for it.